src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1943,6 +1943,110 @@ internal void IncrementPasteOffset()
             this.currentPasteOffset = (this.currentPasteOffset + PasteOffsetStep) % PasteOffsetMax;
         }
 
+        #region [Nodes Info]
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings
+        /// </summary>
+        public bool HasWarnings
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Warning || n.State == ElementState.PersistentWarning); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings with no Geometry
+        /// </summary>
+        public bool HasWarningsWithNoGeometry
+        {
+            get { return Nodes.Any(n => (n.State == ElementState.Warning || n.State == ElementState.PersistentWarning) && !n.Category.StartsWith("Geometry.")); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if workspace run with errors
+        /// </summary>
+        public bool HasErrors
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Error); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if home workspace is displayed with infos
+        /// </summary>
+        public bool HasInfos
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Info); }
+        }
+
+        /// <summary>
+        /// All the Nodes identified as deprecated
+        /// </summary>
+        public IEnumerable<NodeModel> DeprecatedNodes
+        {
+            get
+            {
+                return Nodes.Where(node => node.NodeInfos.Any(nodeInfo => nodeInfo.Message.Contains("This method is deprecated and will be removed")));
+            }
+        }
+
+        /// <summary>
+        /// Indicates if the Workspace have deprecated Nodes
+        /// </summary>
+        public bool AreDeprecatedNodes
+        {
+            get
+            {
+                return DeprecatedNodes.Count() > 0;
+            }
+        }
+
+        /// <summary>
+        /// Indicates if all the deprecated nodes have metadata, bearing on mind if all their ports are connected
+        /// </summary>
+        public bool AllDeprecatedNodesHaveMetadata
+        {
+            get
+            {
+                return DeprecatedNodes.All(node =>
+                node.InPorts.Count() == node.InPorts.Count(inPort => inPort.IsConnected == true) &&
+                node.OutPorts.Count() == node.OutPorts.Count(outPort => outPort.IsConnected == true));
+            }
+        }
+
+        /// <summary>
+        /// Indicates if the workspace has crashes or seriuos errors
+        /// </summary>
+        public bool AreThereCrashesOrSeriuosErrors
```

Comment:
typo, should be `AreThereCrashesOrSeriousErrors`

Reply:
In any case, this name doesn't seem fit as you're also counting Infos and that certainly isn't a crash or serious error.